lib/lhm.rb

```diff
@@ -82,15 +83,26 @@ def cleanup_current_run(run, table_name, options = {})
     Lhm::Cleanup::Current.new(run, table_name, connection, options).execute
   end
 
-  def setup(connection)
-    @@connection = Lhm::Connection.new(connection: connection)
+  # Setups DB connection
+  #
+  # @param [ActiveRecord::Base] connection ActiveRecord Connection
+  # @param [Hash] connection_options Optional options (defaults to: empty hash)
+  # @option options [Boolean] :reconnect_with_consistent_host
+  #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
+  def setup(connection, connection_options = {})
+    @@connection = Connection.new(connection: connection, options: connection_options)
   end
 
-  def connection
+  # Setups DB connection
+  #
+  # @param [Hash] connection_options Optional options (defaults to: empty hash)
+  # @option options [Boolean] :reconnect_with_consistent_host
+  #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
+  def connection(connection_options = {})
```

Comment:
I see what you mean, it can be dangerous. This is part of LHM interface and some gem users might be using it. There are 2 ways this could be addressed:
1) Change the way the method works, by only returning the instance of connection and raising an error if not defined (need to use `Lhm.setup` as a consistent way of setting a connection)
2) Find a way to hot-swap the new connection_options

Any preference?

Reply:
Following the principle of least surprises, the first one seems like it's more surprising if i think about it now.